src/zarr/core/array.py

```diff
@@ -161,6 +161,77 @@ async def get_array_metadata(
 
 @dataclass(frozen=True)
 class AsyncArray:
+    """
+    An asynchronous array class representing a chunked array stored in a Zarr store.
+
+    Parameters
+    ----------
+    metadata : ArrayMetadata
+        The metadata of the array.
+    store_path : StorePath
+        The path to the Zarr store.
+    codec_pipeline : CodecPipeline, optional
+        The codec pipeline used for encoding and decoding chunks, by default None.
+    order : {'C', 'F'}, optional
+        The order of the array data in memory, by default None.
+
+    Attributes
+    ----------
+    metadata : ArrayMetadata
+        The metadata of the array.
+    store_path : StorePath
+        The path to the Zarr store.
+    codec_pipeline : CodecPipeline
+        The codec pipeline used for encoding and decoding chunks.
+    order : {'C', 'F'}
+        The order of the array data in memory.
+
+    Methods
```

Comment:
I think this Methods section can be removed. Usually it's only used if you want to highlight a few methods on a class with many methods. I suspect that we want everything and use sphinx's autodoc to link everything together.

Reply:
okay, thanks!